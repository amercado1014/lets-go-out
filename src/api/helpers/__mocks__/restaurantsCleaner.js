export const restaurantsCleaner = jest.fn();