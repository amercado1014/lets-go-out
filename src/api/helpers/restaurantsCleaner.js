export const restaurantsCleaner = (restaurantsArray) => {
  return restaurantsArray.map(restaurant => {
    const { apiKey, logoUrl, name, streetAddress, city, state, zip, foodTypes, 
      phone, url, deliveryMin, minFreeDelivery,
      offersDelivery, deliveryPrice } = restaurant;
    const hours = Object.keys(restaurant.hours).reduce((hoursObject, day) => {
      if (!hoursObject[day]) {
        hoursObject[day] = restaurant.hours[day].join(',\n');
      }
      return hoursObject;
    }, {});
    
    return ({
      apiKey,
      logoUrl,
      name,
      streetAddress,
      city,
      state,
      zip,
      foodTypes: foodTypes.join(',\n'),
      phone,
      url,
      hours,
      deliveryMin,
      offersDelivery,
      minFreeDelivery,
      deliveryPrice
    });
  });
};
