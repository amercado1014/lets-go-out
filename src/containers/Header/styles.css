.header {
  display: flex;
  justify-content: space-around;
  align-items: center;
  background-color: #afd045;
}

.location {
  font-size: 20px;
}

.location-logo {
  width: 15px;
  height: 15px;
  margin-right: 10px;
}

input {
  height: 25px;
  width: 215px;
}

.header-input {
  font-size: 20px;
  color: black;
  background-color: #afd045;
}

.signin-link {
  text-decoration: none;
  font-size: 20px;
  color: black;
}

.signin-link:hover {
  color: white;
}

.signin-link:active {
  color: purple;
}