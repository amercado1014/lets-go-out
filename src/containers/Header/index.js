import React, { Component } from 'react';
import './styles.css';
import { fetchRestaurantsBySearch 
} from '../../api/apiCalls/fetchRestaurantsBySearch';
import { restaurantsCleaner } from '../../api/helpers/restaurantsCleaner';
import { addRestaurants } from '../../actions/';
import { connect } from 'react-redux';
import * as routes from '../../constants/routes';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export class Header extends Component {
  constructor(props) {
    super(props);

    this.state = {
      searchValue: ''
    };
  }

  handleChange = event => {
    const { value } = event.target;
    this.setState({ searchValue: value });
  }

  handleSubmit = async event => {
    const { searchValue } = this.state;
    event.preventDefault();
    try {
      const restaurants = await fetchRestaurantsBySearch(searchValue);
      const cleanRestaurants = restaurantsCleaner(restaurants);
      this.props.addRestaurants(cleanRestaurants);
    } catch (error) {
      throw error.message;
    }
  }

  render() {
    return (
      <div>
        <h1>{ "Let's Order Out" }</h1>
        <form onSubmit={this.handleSubmit}>
          <input 
            type="text"
            value={this.state.searchValue}
            placeholder="Enter city or zip code"
            onChange={this.handleChange}/>
          <button type="submit">Search</button>
        </form>
        <div className="navigation">
          <Link to={routes.SIGN_UP}>Sign Up</Link>
        </div>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  addRestaurants: restaurants => dispatch(addRestaurants(restaurants))
});

Header.propTypes = {
  addRestaurants: PropTypes.func
};

export default connect(null, mapDispatchToProps)(Header);
