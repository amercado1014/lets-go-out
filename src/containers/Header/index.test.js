import React from 'react';
import { Header } from '../Header';
import { shallow } from 'enzyme';
import { fetchRestaurantsBySearch 
} from '../../api/apiCalls/fetchRestaurantsBySearch';
import { restaurantsCleaner } from '../../api/helpers/restaurantsCleaner';

jest.mock('../../api/apiCalls/fetchRestaurantsBySearch');
jest.mock('../../api/helpers/restaurantsCleaner');

describe('Header', () => {
  let wrapper;
  let mockAddRestaurants;

  beforeEach(() => {
    mockAddRestaurants = jest.fn();
    wrapper = shallow(<Header addRestaurants={mockAddRestaurants} />);
  });

  it('should math the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state of an empty string for searchValue', () => {
    expect(wrapper.state('searchValue')).toEqual('');
  });

  it('should update the state when handleChange is called', () => {
    const mockEvent = { target: { value: 'Hello World' } };
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('searchValue')).toEqual('Hello World');
  });

  it('should call fetchRestaurantsBySearch on handleSubmit', () => {
    wrapper.setState({ searchValue: 80227 });
    const expected = wrapper.state("searchValue");
    const mockEvent = { preventDefault: jest.fn() };
    wrapper.instance().handleSubmit(mockEvent);
    expect(fetchRestaurantsBySearch).toHaveBeenCalledWith(expected);
  });

  it('should call restaurantsCleaner on handleSubmit', () => {
    const mockEvent = { preventDefault: jest.fn() };
    wrapper.instance().handleSubmit(mockEvent);
    expect(restaurantsCleaner).toHaveBeenCalled();
  });

  it.skip('should call addRestaurants on handleSubmit', () => {
    const mockEvent = { preventDefault: jest.fn() };
    wrapper.instance().handleSubmit(mockEvent);
    expect(mockAddRestaurants).toHaveBeenCalled();
  });
});
