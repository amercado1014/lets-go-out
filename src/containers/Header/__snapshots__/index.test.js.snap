// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should math the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    addRestaurants={[Function]}
    history={
        Object {
            "push": [Function],
          }
    }
    locationOff={[Function]}
    restaurants={
        Array [
            Object {
              "apiKey": "aa1163f8eb68a8f7144247b91fbcb2f39e65e39821ec08a9",
              "city": "Lakewood",
              "deliveryMin": 12,
              "foodTypes": "Pizza",
              "hours": Object {
                "Friday": "11:00 AM - 3:00 AM",
                "Monday": "11:00 AM - 3:00 AM",
                "Saturday": "11:00 AM - 3:00 AM",
                "Sunday": "11:00 AM - 3:00 AM",
                "Thursday": "11:00 AM - 3:00 AM",
                "Tuesday": "11:00 AM - 3:00 AM",
                "Wednesday": "11:00 AM - 3:00 AM",
              },
              "logoUrl": "https://eatstreet-static.s3.amazonaws.com/assets/images/restaurant_logos/big-daddys-pizza-lakewood-21135_1407868711096.png",
              "minFreeDelivery": 0,
              "name": "Big Daddy's Pizza - Lakewood",
              "offersDelivery": false,
              "phone": "(303) 736-6090",
              "state": "CO",
              "streetAddress": "850 Wadsworth Blvd",
              "url": "https://eatstreet.com/lakewood-co/restaurants/big-daddys-pizza-lakewood",
              "zip": "80214",
            },
            Object {
              "apiKey": "aa1163f8eb68a8f7144247b91fbcb2f39e65e39821ec08a9",
              "city": "Lakewood",
              "deliveryMin": 12,
              "foodTypes": "Pizza",
              "hours": Object {
                "Friday": "11:00 AM - 3:00 AM",
                "Monday": "11:00 AM - 3:00 AM",
                "Saturday": "11:00 AM - 3:00 AM",
                "Sunday": "11:00 AM - 3:00 AM",
                "Thursday": "11:00 AM - 3:00 AM",
                "Tuesday": "11:00 AM - 3:00 AM",
                "Wednesday": "11:00 AM - 3:00 AM",
              },
              "logoUrl": "https://eatstreet-static.s3.amazonaws.com/assets/images/restaurant_logos/big-daddys-pizza-lakewood-21135_1407868711096.png",
              "minFreeDelivery": 0,
              "name": "Big Daddy's Pizza - Lakewood",
              "offersDelivery": false,
              "phone": "(303) 736-6090",
              "state": "CO",
              "streetAddress": "850 Wadsworth Blvd",
              "url": "https://eatstreet.com/lakewood-co/restaurants/big-daddys-pizza-lakewood",
              "zip": "80214",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="logo"
          className="logo"
          src="logo.png"
/>,
        <p
          className="location"
>
          <img
                    alt="logo"
                    className="location-logo"
                    src="location.png"
          />
          Lakewood
          , 
          CO
</p>,
        <form
          onSubmit={[Function]}
>
          <input
                    className="header-input"
                    onChange={[Function]}
                    placeholder="Enter city or zip code"
                    type="text"
                    value=""
          />
</form>,
        <div
          className="navigation"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="header-link"
                    to="/signin"
          >
                    Sign In
          </NavLink>
</div>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "logo",
          "className": "logo",
          "src": "logo.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="location-logo"
              src="location.png"
/>,
            "Lakewood",
            ", ",
            "CO",
          ],
          "className": "location",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "location-logo",
              "src": "location.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "Lakewood",
          ", ",
          "CO",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="header-input"
            onChange={[Function]}
            placeholder="Enter city or zip code"
            type="text"
            value=""
/>,
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "header-input",
            "onChange": [Function],
            "placeholder": "Enter city or zip code",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="header-link"
            to="/signin"
>
            Sign In
</NavLink>,
          "className": "navigation",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Sign In",
            "className": "header-link",
            "to": "/signin",
          },
          "ref": null,
          "rendered": "Sign In",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="logo"
            className="logo"
            src="logo.png"
/>,
          <p
            className="location"
>
            <img
                        alt="logo"
                        className="location-logo"
                        src="location.png"
            />
            Lakewood
            , 
            CO
</p>,
          <form
            onSubmit={[Function]}
>
            <input
                        className="header-input"
                        onChange={[Function]}
                        placeholder="Enter city or zip code"
                        type="text"
                        value=""
            />
</form>,
          <div
            className="navigation"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="header-link"
                        to="/signin"
            >
                        Sign In
            </NavLink>
</div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "className": "logo",
            "src": "logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="location-logo"
                src="location.png"
/>,
              "Lakewood",
              ", ",
              "CO",
            ],
            "className": "location",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "location-logo",
                "src": "location.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "Lakewood",
            ", ",
            "CO",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="header-input"
              onChange={[Function]}
              placeholder="Enter city or zip code"
              type="text"
              value=""
/>,
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "header-input",
              "onChange": [Function],
              "placeholder": "Enter city or zip code",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="header-link"
              to="/signin"
>
              Sign In
</NavLink>,
            "className": "navigation",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Sign In",
              "className": "header-link",
              "to": "/signin",
            },
            "ref": null,
            "rendered": "Sign In",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
